web/datapipeline/hosts_projector.go

```diff
@@ -128,67 +129,22 @@ func parseCloudData(provider string, metadata interface{}) *entities.AzureCloudD
 }
 
 func parseAzureCloudData(metadata interface{}) entities.AzureCloudData {
-	var cloudData entities.AzureCloudData
-	metadataMap, ok := metadata.(map[string]interface{})
-	if !ok {
-		log.Errorf("can't decode Azure metadata")
-		return cloudData
-	}
-
-	computeInfo, ok := metadataMap["compute"].(map[string]interface{})
-	if !ok {
-		log.Errorf("can't decode compute metadata")
-		return cloudData
-	}
-
-	if _, ok := computeInfo["name"]; ok {
-		cloudData.VMName = computeInfo["name"].(string)
-	}
-
-	if _, ok := computeInfo["name"]; ok {
-		cloudData.Offer = computeInfo["offer"].(string)
-	}
-
-	if _, ok := computeInfo["resourceGroupName"]; ok {
-		cloudData.ResourceGroup = computeInfo["resourceGroupName"].(string)
-	}
-
-	if _, ok := computeInfo["sku"]; ok {
-		cloudData.SKU = computeInfo["sku"].(string)
-	}
+	var azureMetadata cloud.AzureMetadata
 
-	if _, ok := computeInfo["vmSize"]; ok {
-		cloudData.VMSize = computeInfo["vmSize"].(string)
-	}
-
-	if _, ok := computeInfo["location"]; ok {
-		cloudData.Location = computeInfo["location"].(string)
-	}
-
-	osProfile, ok := computeInfo["osProfile"].(map[string]interface{})
-	if !ok {
-		log.Errorf("can't decode os profile")
-		return cloudData
-	}
-
-	if _, ok := osProfile["adminUsername"]; ok {
-		cloudData.AdminUsername = osProfile["adminUsername"].(string)
+	err := mapstructure.Decode(metadata, &azureMetadata)
+	if err != nil {
+		log.Errorf("can't decode azure metadata: %s", err)
+		return entities.AzureCloudData{}
 	}
 
-	storageProfile, ok := computeInfo["storageProfile"].(map[string]interface{})
-	if !ok {
-		log.Errorf("can't decode storage profile")
-		return cloudData
+	return entities.AzureCloudData{
+		VMName:          azureMetadata.Compute.Name,
```

Comment:
mmm, I have some doubts about this.
Could we have the scenario where some of this fields don't exist and we have some other issue.
Or they are automatically filled with empty data?
The added test doesn't cover this situation at least

Reply:
it should work because the fields are not pointers so it will take the golang "zero value".

Reply:
@arbulu89 I've added a test, i still believe it is a bit anemic and we are basically testing that golang works but we can keep it as safety helmet